modules/hivemq-edge-module-modbus/src/main/java/com/hivemq/edge/adapters/modbus/ModbusAdapterConfig.java

```diff
@@ -100,9 +103,31 @@ public static class Subscription extends AbstractProtocolAdapterConfig.Subscript
                            description = "Define the start and end index values for your memory addresses")
         private @NotNull AddressRange addressRange;
 
+        @JsonCreator
+        public Subscription(
+                @JsonProperty("destination") @Nullable final String destination,
+                @JsonProperty("qos") final int qos,
+                @JsonProperty("addressRange") @NotNull final AddressRange addressRange) {
+            super(destination, qos);
+            this.addressRange = addressRange;
+        }
+
         public @NotNull AddressRange getAddressRange() {
             return addressRange;
         }
+
+        @Override
+        public boolean equals(final Object o) {
+            if (this == o) return true;
+            if (!(o instanceof Subscription)) return false;
+            Subscription that = (Subscription) o;
+            return Objects.equals(addressRange, that.addressRange);
```

Comment:
shouldnt be QoS and Destination also be part of equals and hashcode? If there is a reason to not have them inside please make a comment so we avoid a future bug.